### PR TITLE
TEST-198 Refactor JsonWebToken API in MP TCK Runner JWT Auth 

### PR DESCRIPTION
**Pull request by AlanRoth**
Made a profile to include the moved module  from MP-TCK to Payara-server.

Relies on #4054 in Payara
https://github.com/payara/Payara/pull/4054

### CODE REVIEW

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml [hunk 1]
```diff
@@ -52,12 +52,31 @@
     <packaging>jar</packaging>
 
     <name>MicroProfile JWT Auth TCK Arquillian Extension</name>
-
+
```
Comment (cubastanley) [inline MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: Tabzez

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml [hunk 2]
```diff
@@ -57,6 +57,25 @@
...
+                <property>
+                    <name>payara.version</name>
+                    <value>/4.+/</value>
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>fish.payara.microprofile.jwt-auth</groupId>
+                    <artifactId>microprofile-jwt-auth</artifactId>
+                    <version>{payara.version}</version>
```
Comment (pdudits) [inline MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: missing `$`?

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml [hunk 3]
```diff
@@ -57,6 +57,42 @@
...
+                <property>
+                    <name>payara.version</name>
+                    <value>/5\.([1-9][9][3-4]|[2-9][0-9][0-9])/</value>
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>fish.payara.microprofile.jwt-auth</groupId>
+                    <artifactId>microprofile-jwt-auth</artifactId>
+                    <version>{$payara.version}</version>
```
Comment (pdudits) [inline MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: Almost there. Dollar sign should be outside of braces.

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml [hunk 4]
```diff
@@ -57,6 +57,42 @@
         <!-- JSON Dependencies -->
         <json.version>1.1.2</json.version>
     </properties>
+    
+    <profiles>
+       <!-- This profile will exclude the TokenParser class and use the one in Payara Server instead. For versions 5.193+ -->
+        <profile>
+            <id>payara5</id>
```
Comment (jGauravGupta) [inline MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: `payara5` is not a suitable name for this profile as activated based on regex only for 5.193+ version.

**Comment by jGauravGupta** [comment]
Hi @AlanRoth / @MattGill98 ,

Instead of moving `TokenParser` class to Server, Can't we have 2 classes in `tck-arquillian-extension` itself? One class for an older version and other for 5.193+. A class should be compiled and included based on `maven-regex-profile-activation-extension`. I can't find any use of `TokenParser` class on the server except used by TCK Runner.

**Comment by MattGill98** [comment]
Hi @jGauravGupta,

That is an option. You're right in that the `TokenParser` class isn't used anywhere except for the TCK runners, but I think it should follow a pattern similar to the mock tracer in OpenTracing, since (as far as I can remember from when I looked at this) that would prevent us needing to worry about future changes to the class again. The other problem is that running the TCK runners suite currently requires that you have an internal Payara module built, which is quite tedious for testing with.

Both options are ultimately viable so I won't fight to the death for this way, but I do slightly favour it.

What do you think?

**Comment by jGauravGupta** [comment]
Found the Mock Tracer here: 
https://github.com/payara/Payara/blob/56be23561117a67d1cfe3126d01818c36ae07718/nucleus/payara-modules/opentracing-adapter/src/main/java/fish/payara/opentracing/OpenTracingService.java#L102

IMO, `TokenParser` class should also have Mock prefix to distinguish it with `JwtTokenParser` class.

**Comment by MattGill98** [comment]
Sounds good to me!

**Pull request merged**